File: cookbook/php/getTargetFromUnicity.php
<?php

/**
 * Ce script permet de récuperer une cible grace à son caractère d'unicité.
 *
 * @package cookbook
 */

require 'utils.php';

/**
 * Variable contenant les configurations pour se connecter à l'API
 *
 * @var array
 */
$configs = parse_ini_file("config.ini");
/**
 * Caractère d'unicité qui vous permet d'identifier la cible que vous recherchez
 *
 * @var string
 */
$unicity = 'test@test.com';

// On trouve l'adresse pour la requete
$url = $configs['url'] . 'targets?unicity='. $unicity;
$con = connect($url, $configs['xKey'], null, 'GET');

// Verification des reponses
if ($con['result'] == true) {
	echo $con['result'];
}
else {
	// Affichage de l'erreur
	echo 'Error : ' . $con['info']['http_code'];
}

?>

File: python/GetTargetFromUnicity.py
"""
Ce script permet de récupérer une Cible grace à un des ses caractères d'unicité.
"""

import requests
import Utils

def get_target(config, unicity):
    """Récupération de la cible"""
    url = config['API']['url'] + 'targets?unicity=' + unicity

    # Connexion à l'API
    headers = Utils.create_headers(config['API']['xKey'], 0)
    req = requests.get(url, headers=headers)
    req.raise_for_status()
    print(req.text)

def main():
    """
    Pensez à modifier le caractère d'unicité.
    """
    unicity = "test@test.com"
    config = Utils.load_config()
    get_target(config, unicity)

if __name__ == '__main__':
    main()
